src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/ApproverCaseloadService.kt

```diff
@@ -3,24 +3,141 @@ package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.caseload
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.ApprovalCase
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.LicenceSummaryApproverView
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.ProbationPractitioner
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.PrisonApproverService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.convertToTitleCase
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerSearchApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerSearchPrisoner
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.CommunityApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.OffenderDetail
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.ProbationSearchApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.User
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+import java.time.LocalDate
+import java.time.LocalDateTime
 
 @Service
 class ApproverCaseloadService(
   private val prisonApproverService: PrisonApproverService,
+  private val probationSearchApiClient: ProbationSearchApiClient,
+  private val prisonerSearchApiClient: PrisonerSearchApiClient,
+  private val communityApiClient: CommunityApiClient,
 ) {
 
-  fun getApprovalNeeded(prisons: List<String>): List<LicenceSummaryApproverView> {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun getApprovalNeeded(prisons: List<String>): List<ApprovalCase> {
     val filteredPrisons = prisons.filterNot { it == "CADM" }
     val licences = prisonApproverService.getLicencesForApproval(filteredPrisons)
     if (licences.isEmpty()) {
       return emptyList()
     }
-    return licences
+
+    val nomisIds = licences.mapNotNull { it.nomisId }
+    val deliusRecords = probationSearchApiClient.searchForPeopleByNomsNumber(nomisIds)
+    val deliusNomisIds = deliusRecords.mapNotNull { it.otherIds.nomsNumber }
+    val nomisRecords = prisonerSearchApiClient.searchPrisonersByNomisIds(deliusNomisIds)
+
+    val prisonerRecord: List<Triple<PrisonerSearchPrisoner, OffenderDetail?, LicenceSummaryApproverView?>> =
+      nomisRecords.map {
+        Triple(it, it.getDeliusRecord(deliusRecords), it.getLicenceSummary(licences))
+      }
+
+    val caseload = prisonerRecord.map { (nomisRecord, deliusRecord, licenceSummary) ->
+      val releaseDate = when (licenceSummary) {
```

Comment:
Umming and ahhing about this as although this is in the frontend, a licence summary wouldn't be null at this point as it is used to retrieve the `nomisIds` which drive the searches in NOMIS and Delius. Don't think this is needed but tried to match what is on the frontend.

Reply:
yes, totally right - this was something in hindsight I thought I should have removed. I was copying over from the existing implementation. 

I think remove it?